dariobmibar/src/main/java/com/labstyle/dariobmibar/DarioBMIDotIndicator.kt

```diff
@@ -0,0 +1,127 @@
+package com.labstyle.dariobmibar
+
+import android.content.Context
+import android.icu.lang.UCharacter.GraphemeClusterBreak.V
+import android.util.AttributeSet
+import android.util.Log
+import android.view.View
+import android.widget.RelativeLayout
+import android.widget.TextView
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.content.ContextCompat
+import java.util.Locale
+
+class DarioBMIDotIndicator @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0,
+    defStyleRes: Int = 0
+): RelativeLayout(context, attrs, defStyle, defStyleRes) {
+    private var bmi = 21f
+    private var unitLabel = "BMI"
+
+    init {
+        val attributes = context.obtainStyledAttributes(attrs, R.styleable.DarioBMIBar)
+        bmi = attributes.getFloat(R.styleable.DarioBMIBar_bmi, bmi)
+        unitLabel = attributes.getString(R.styleable.DarioBMIBar_unitLabel) ?: unitLabel
+        attributes.recycle()
+
+        inflate(context, R.layout.dario_bmi_dot_indicator, this)
+
+        setBMIUnitLabel(unitLabel)
+        setBMIValue(bmi)
+    }
+
+    fun setBMIUnitLabel(label: String) {
+        findViewById<TextView>(R.id.bmiText)?.text = label
+    }
+
+    fun setBMIValue(value: Float) {
+        bmi = value
+
+
+        val parent = findViewById<ConstraintLayout>(R.id.topBarsLayout)
+        val constraintSet = ConstraintSet()
+        constraintSet.clone(parent)
+
+        var hBias = calculateHBias(bmi)
+
+        if (hBias > 0.985f) {
+            hBias = 0.985f
+        }
+        if (hBias < 0.01f) {
+            hBias = 0.01f
+        }
+
+        constraintSet.setHorizontalBias(R.id.selectionLineIndicator, hBias)
+        constraintSet.clear(R.id.bmiText, ConstraintSet.START)
+        constraintSet.clear(R.id.bmiText, ConstraintSet.END)
+
+        if (hBias > 0.85f) {
+            constraintSet.connect(
+                R.id.bmiText,
+                ConstraintSet.END,
+                R.id.selectionLineIndicator,
+                ConstraintSet.START
+            )
+        } else if (hBias < 0.08f) {
+            // Position bmiText to the left
+            constraintSet.connect(
+                R.id.bmiText,
+                ConstraintSet.START,
+                R.id.selectionLineIndicator,
+                ConstraintSet.END
+            )
+        } else {
+            constraintSet.connect(
+                R.id.bmiText,
+                ConstraintSet.START,
+                R.id.selectionLineIndicator,
+                ConstraintSet.START
+            )
+            constraintSet.connect(
+                R.id.bmiText,
+                ConstraintSet.END,
+                R.id.selectionLineIndicator,
+                ConstraintSet.END
+            )
+        }
+
+        constraintSet.applyTo(parent)
+
+        val circleBg =
```

Comment:
it's the same drawable for all cases here.

Reply:
yes